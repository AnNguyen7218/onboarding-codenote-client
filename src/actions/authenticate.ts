import { createAction } from 'redux-actions';

export const actionTypes = {
  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
};

export const userHasAuthenticated = createAction(actionTypes.LOGIN_SUCCESS);
