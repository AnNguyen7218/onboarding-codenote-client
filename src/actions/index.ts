import {LOGGED_IN} from './type'

export const userHasAuthenticated = (value) => ({
  type: LOGGED_IN,
  payload: value
})
