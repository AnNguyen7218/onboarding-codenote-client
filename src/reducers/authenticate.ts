const initialState = {
  isAuthenticated: false,
};

export default function authenticate(state, action) {
};
