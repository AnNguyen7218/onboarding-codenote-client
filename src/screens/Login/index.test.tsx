import React from 'react';
import createStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import Login from './index';
import { userHasAuthenticated } from '../../actions/authenticate'

test('Login component renders without crashing', () => {
  let store = createStore()();
  const wrapper = shallow(<Login store={store} />).dive();
  expect(wrapper).toMatchSnapshot();
});

test('Login component - checks action on dispatching ', () => {
  let action;
  let store = createStore()();
  store.dispatch(userHasAuthenticated(true))
  action = store.getActions()
  expect(action[0].type).toBe("LOGIN_SUCCESS")
});

test('handleChange will update the state', () => {
  let store = createStore()();
  const event = {
          target : {
            id: 'email',
            value: "cunmap@mailinator.com"
          }
        }
  const wrapper = shallow(<Login store={store} />).dive();
  wrapper.instance().handleChange(event);
  expect(wrapper.state().email).toBe(event.target.value);
});

test('handleSubmit will update the state', () => {
  let store = createStore()();
  const event = {
    preventDefault: () => {}
  };
  const wrapper = shallow(<Login store={store} />).dive();
  wrapper.instance().handleSubmit(event);
  expect(wrapper.state().isLoading).toBe(true);
});