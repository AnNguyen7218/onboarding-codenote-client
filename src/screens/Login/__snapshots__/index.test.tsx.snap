// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders without crashing 1`] = `
<div
  className="Login"
>
  <form
    onSubmit={[Function]}
  >
    <FormGroup
      bsClass="form-group"
      bsSize="large"
      controlId="email"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Email
      </ControlLabel>
      <FormControl
        autoFocus={true}
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        type="email"
        value=""
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      bsSize="large"
      controlId="password"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Password
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        type="password"
        value=""
      />
    </FormGroup>
    <LoaderButton
      block={true}
      bsSize="large"
      disabled={true}
      isLoading={false}
      loadingText="Logging in ..."
      text="Login"
      type="submit"
    />
  </form>
</div>
`;
